src/mcts/search.cc

```diff
@@ -1330,13 +1343,15 @@ void SearchWorker::DoBackupUpdateSingleNode(
 
     // A non-winning terminal move needs all other moves to be similar.
     auto all_losing = true;
+    auto found_tb = n->IsTbTerminal();
     float losing_m = 0.0f;
     if (can_convert && v <= 0.0f) {
       for (const auto& edge : p->Edges()) {
         const auto WL = edge.GetWL();
         can_convert = can_convert && edge.IsTerminal() && WL <= 0.0f;
         if (!can_convert) break;
         all_losing = all_losing && WL < 0.0f;
+        found_tb = found_tb || edge.IsTbTerminal();
```

Comment:
Does it mean that if there is a short non-tb win, and long tb win, it will mark the node as tb-win, and will show 100 additional moves in mate score?

Reply:
This code path is not for winning `v > 0`, so the behavior here is if the position has all losing moves, assume the tb loss will be played because it's probably longer than an actual loss.

For found winning `v > 0`, the first one found will cause the parent to convert to loss matching the found move's TB-ness-or-not, which indeed can overestimate the moves left because a shorter win might be found that we won't explore from the current search as future PickNodeToExtend will stop at the now-terminal parent loss.
https://github.com/LeelaChessZero/lc0/blob/f6306b350a59be774d2194b0146a3f7657243729/src/mcts/search.cc#L1348-L1351